tsdb/engine/tsm1/compact.go

```diff
@@ -1417,13 +1429,18 @@ type tsmBatchKeyIterator struct {
 	overflowErrors int
 }
 
+// AppendError - store unique errors in the order of first appearance,
+// up to a limit of maxErrors.  If the error is unique and stored, return true.
 func (t *tsmBatchKeyIterator) AppendError(err error) bool {
-	if t.maxErrors > len(t.errs) {
+	s := err.Error()
+	if _, ok := t.errSet[s]; ok {
+		return true
```

Comment:
Not sure if we should count how many duplicate errors we received. Might be useful so you can tell the difference between a file that had a single error and a file that had 25 of the same error. I'll approve without this. If you want to add it, I can re-review after the change.

Reply:
My sens is that replicated errors don't need to be counted; it just means the query or whatever was re-submitted.  I'll merge as is.  Thanks!